filters/include/pcl/filters/spherical_voxel_grid.h

```diff
@@ -0,0 +1,357 @@
+        */
+      inline unsigned int
+      getMinimumPointsNumberPerVoxel () { return (min_points_per_voxel_); }
+
+      /** \brief Return the maximum radius of a point from the ceter,
+        * i.e. the radius of the spherical voxel grid (after filtering)
+        */
+      inline float
+      getMaximumRadius () { return (max_radius_); }
+
+      /** \brief Set the cartesian origin used to build the spherical grid.
+        * \param[in] origin the cartesian origin
+        */
+      inline void
+      setOrigin (const Eigen::Vector4f& origin) { filter_origin_ = origin; }
+
+      /** \brief Get the cartesian origin used to build the spherical grid.
+        */
+      inline Eigen::Vector3f
+      getOrigin () { return (filter_origin_.head<3>()); }
+
+      /** \brief Provide the name of the field to be used for filtering data. In conjunction with  \a setFilterLimits,
+        * points having values outside this interval will be discarded.
+        * \param[in] field_name the name of the field that contains values used for filtering
+        */
+      inline void
+      setFilterFieldName (const std::string &field_name) { filter_field_name_ = field_name; }
+
+      /** \brief Get the name of the field used for filtering. */
+      inline std::string const
+      getFilterFieldName () { return (filter_field_name_); }
+
+      /** \brief Set the field filter limits. All points having field values outside this interval will be discarded.
+        * \param[in] limit_min the minimum allowed field value
+        * \param[in] limit_max the maximum allowed field value
+        */
+      inline void
+      setFilterLimits (const double &limit_min, const double &limit_max)
+      {
+        filter_limit_min_ = limit_min;
+        filter_limit_max_ = limit_max;
+      }
+
+      /** \brief Get the field filter limits (min/max) set by the user. The default values are -FLT_MAX, FLT_MAX.
+        * \param[out] limit_min the minimum allowed field value
+        * \param[out] limit_max the maximum allowed field value
+        */
+      inline void
+      getFilterLimits (double &limit_min, double &limit_max)
+      {
+        limit_min = filter_limit_min_;
+        limit_max = filter_limit_max_;
+      }
+
+      /** \brief Set to true if we want to return the data outside the interval specified by setFilterLimits (min, max).
+        * Default: false.
+        * \param[in] limit_negative return data inside the interval (false) or outside (true)
+        */
+      inline void
+      setFilterLimitsNegative (const bool limit_negative) { filter_limit_negative_ = limit_negative; }
+
+      /** \brief Get whether the data outside the interval (min/max) is to be returned (true) or inside (false).
+        * \param[out] limit_negative true if data \b outside the interval [min; max] is to be returned, false otherwise
+        */
+      inline void
+      getFilterLimitsNegative (bool &limit_negative) { limit_negative = filter_limit_negative_; }
```

Comment:
Why not returning?

Reply:
Preserving consistent interface with `VoxelGrid`